bin/weldr/src/weldr.rs

```diff
@@ -195,7 +192,6 @@ impl DiskResolver {
             .map_err(|e| Error::NotFound(format!("catalog path not found ({})", e)))?;
         let base_paths = vec![
             catalog_path.join("p"),
-            catalog_path.join("p").join("48"),
```

Comment:
Why was this removed? Is the latest catalog not using a "48" folder anymore?

Reply:
> Why was this removed? Is the latest catalog not using a "48" folder anymore?

The 48 folder is the high resolution primitives folder. There are also low res primitives in the 8 folder. Normal resolution primitives are just in the p folder. Some applications support switching primitives. Just sticking with standard resolution primitives seems like a reasonable default for gltf export for now. Application code can always write their own resolver.

> Did I get the reasoning correctly?

Yeah. It's a tradeoff between memory usage and per object overhead. The extension check is a bit of a hack, but I haven't run into any issues doing it that way while developing https://github.com/ScanMountGoat/ldr_tools_blender. It's probably worth revisiting at some point.

Older LDraw applications use immediate mode rendering on each triangle or quad line in each file with push/pop matrix commands. This made sense on older hardware with limited memory but performs poorly even on newer GPUs. All of the modern applications and importers I've looked at merge all the geometry for each "part". 

A more important reason to merge like this is that apparently subparts sometimes use shearing matrices. This isn't a problem for gltf but won't work in applications that decompose transforms into translation, rotation, and scale. This can be resolved by merging all the geometry at the part level. You can see an example if you search for "shear" on the readme of https://github.com/cuddlyogre/ExportLDraw.